File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp" // MODIFY MESSAGE TYPE IF NEEDED

using namespace std::chrono_literals;
     
class RobotNewsStationNode: public rclcpp::Node { //
    public:
        RobotNewsStationNode() : Node("robot_news_station") { // MODIFY NAME
            publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10); 
            timer_ = this->create_wall_timer(500ms, std::bind(&RobotNewsStationNode::publishNews, this));
            RCLCPP_INFO(this->get_logger(), "Robot News Station node has been started from CPP!");
        }
     
    private:
        void publishNews() {
            auto message = example_interfaces::msg::String();
            message.data = "Hello from Robot News Station from CPP!"; // MODIFY MESSAGE CONTENT IF NEEDED
            RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
            publisher_->publish(message);
        }
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_ ;
};
     
int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
     
     
class RobotNewsStation(Node):
    def __init__(self):
        super().__init__("robot_news_station") # node name
        self.publisher_ = self.create_publisher(String, "robot_news", 10)  # create publisher
        self.timer_ = self.create_timer(0.5, self.publish_news)  # create timer to call publish_news every 0.5 seconds
        self.get_logger().info("Robot News Station has been started from python!")# log info message
     
    def publish_news(self): # function to publish news
        msg = String() # create a String message
        msg.data = "Hello, this is your robot news station from python!"
        self.publisher_.publish(msg) # publish the msg
     
def main(args=None):
    rclpy.init(args=args) # initialize rclpy
    node = RobotNewsStation() # create node instance
    rclpy.spin(node) # keep node alive to listen to messages
    rclpy.shutdown() # shutdown ROS2
     
if __name__ == "__main__":
    main()
